easybuild/easyblocks/m/mathematica.py

```diff
@@ -113,35 +128,49 @@ def post_install_step(self):
         """Activate installation by using activation key, if provided."""
         if self.cfg['activation_key']:
             # activation key is printed by using '$ActivationKey' in Mathematica, so no reason to keep it 'secret'
-            self.log.info("Activating installation using provided activation key '%s'." % self.cfg['activation_key'])
+            self.log.info(
+                "Activating installation using provided activation key '%s'." % self.cfg['activation_key'])
             qa = [
-                (r"\(enter return to skip Web Activation\):", self.cfg['activation_key']),
+                (r"\(enter return to skip Web Activation\):",
+                 self.cfg['activation_key']),
                 (r"In\[1\]:= ", 'Quit[]'),
             ]
             no_qa = [
                 '^%s %s .*' % (self.name, self.version),
                 '^Copyright.*',
             ]
-            run_shell_cmd(os.path.join(self.installdir, 'bin', 'math'), qa_patterns=qa, qa_wait_patterns=no_qa)
+            run_shell_cmd(os.path.join(self.installdir, 'bin', 'math'),
+                          qa_patterns=qa, qa_wait_patterns=no_qa)
         else:
-            self.log.info("No activation key provided, so skipping activation of the installation.")
+            self.log.info(
+                "No activation key provided, so skipping activation of the installation.")
 
         super(EB_Mathematica, self).post_install_step()
 
     def sanity_check_step(self):
         """Custom sanity check for Mathematica."""
-        custom_paths = {
-            'files': ['bin/mathematica'],
-            'dirs': ['AddOns', 'Configuration', 'Documentation', 'Executables', 'SystemFiles'],
-        }
+
+        custom_paths = {}
+
+        # in 14.1 we can see rebranding from Mathematica to Wolfram
+        # so GUI launcher is now called "wolframnb"
+        if LooseVersion(self.version) >= LooseVersion("14.1"):
+            app_launcher = 'wolframnb'
+        else:
+            app_launcher = 'mathematica'
+
+        custom_commands = ['echo \'Quit[0]\' | WolframKernel']
+        custom_paths['files'] = [f"bin/{app_launcher}"]
+        custom_paths['dirs'] = ['AddOns', 'Configuration',
+                                'Documentation', 'Executables', 'SystemFiles']
+
         if LooseVersion(self.version) >= LooseVersion("11.3.0"):
             custom_paths['files'].append('Executables/wolframscript')
         elif LooseVersion(self.version) >= LooseVersion("11.0.0"):
             custom_paths['files'].append('bin/wolframscript')
 
-        custom_commands = ['mathematica --version']
-
-        super(EB_Mathematica, self).sanity_check_step(custom_paths=custom_paths, custom_commands=custom_commands)
+        super(EB_Mathematica, self).sanity_check_step(
+            custom_paths=custom_paths, custom_commands=custom_commands)
```

Comment:
a couple of lines break unnecessarily here (eb source code is set to allow 120 character width), and in this particular case, it causes a merge conflict that needs to be resolved